app/views/assignments/_form.html.haml

```diff
@@ -4,8 +4,13 @@
     .col-2.offset-3
       .form-group
         = f.label :user_id
-        = f.collection_select :user_id, @users, :id, :last_name,
-                              {}, class: 'form-control custom-select'
+        - if @current_user.admin_in?(@roster)
+          = f.collection_select :user_id, @users, :id, :last_name,
+            { selected: @assignment.user&.id || @current_user.id }, class: 'form-control custom-select'
+        - else
+          = f.collection_select :user_id, @users, :id, :last_name,
+            { disabled: @users.reject { |u| u == @current_user }.pluck(:id), selected: @current_user.id},
+            class: 'form-control custom-select'
```

Comment:
I don't think there's much point to a select box that will only ever have one option you can select

```suggestion
          = f.hidden_field :user_id, value: @current_user.id
          = f.text_field :user_name, value: @current_user.last_name, disabled: true, class: 'form-control'
```